### PR TITLE
Feature/#2013 user profile

### PR DESCRIPTION
**Pull request by aminbksi**

### CODE REVIEW

File: src/scenes/widgets/pages/ProfileWidget/components/templates/UserProfileView/UserProfileView.tsx [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+                  }}
+                  size="small"
+                />
+                <Button
+                  label={t('actions.highFive')}
+                  onClick={() => {
+                    if (userProfile?.uuid) {
+                      unityStore.sendHighFive(userProfile.uuid);
+                    }
+                  }}
```
Comment (dmitry-yudakov) [inline src/scenes/widgets/pages/ProfileWidget/components/templates/UserProfileView/UserProfileView.tsx:90]: @aminbksi Perhaps it's better be in `handleSendHighFive` function like for the rest of the handlers?

File: src/scenes/widgets/pages/ProfileWidget/components/templates/MyProfileEdit/components/AvatarForm/AvatarForm.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+const AvatarForm: FC = () => {
+  const {widgetStore} = useStore();
+  const {profileStore} = widgetStore;
+  const {userProfile} = profileStore;
+
+  const theme = useTheme();
+
+  const [image, setImage] = useState<File>();
+  const [imageError, setImageError] = useState<boolean>(false);
```
Comment (Daboo1998) [inline src/scenes/widgets/pages/ProfileWidget/components/templates/MyProfileEdit/components/AvatarForm/AvatarForm.tsx:20]: `<boolean>` is redundant here

File: src/scenes/widgets/pages/ProfileWidget/components/templates/UserProfileView/UserProfileView.tsx [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+        {showUserInteractions && (
+          <>
+            <Button label={t('actions.flyTo')} onClick={handleFlyToUser} size="small" />
+            {userProfile?.status !== UserStatusEnum.DO_NOT_DISTURB && (
+              <>
+                <Button
+                  label={t('actions.grabTable')}
+                  onClick={() => {
+                    grabATable();
+                  }}
```
Comment (Daboo1998) [inline src/scenes/widgets/pages/ProfileWidget/components/templates/UserProfileView/UserProfileView.tsx:81]:
```suggestion
                  onClick={grabATable}
```

File: src/scenes/widgets/stores/ProfileStore/ProfileStore.ts [hunk 4]
```diff
@@ -26,19 +27,26 @@ const ProfileStore = types.compose(
...
       profileFetchRequest: types.optional(RequestModel, {}),
       createTableRequest: types.optional(RequestModel, {}),
+      avatarRequest: types.optional(RequestModel, {}),
       inviteToTableRequest: types.optional(RequestModel, {}),
       editProfileRequest: types.optional(RequestModel, {}),
       userOwnedSpacesRequest: types.optional(RequestModel, {}),
       editAvatarDialog: types.optional(DialogModel, {}),
-      userInitiativesRequest: types.optional(RequestModel, {})
+      userInitiativesRequest: types.optional(RequestModel, {}),
+      isProfileEdit: false
```
Comment (Daboo1998) [inline src/scenes/widgets/stores/ProfileStore/ProfileStore.ts:36]: Maybe `isEditingProfile`, something more like a verb

**Review by Daboo1998 — CHANGES_REQUESTED** [review]

File: src/core/models/OnlineUsersList/OnlineUsersList.ts [hunk 5]
```diff
@@ -35,6 +35,11 @@ const OnlineUsersList = types
     setSearchQuery(query: string) {
       self.searchQuery = query;
     }
+  }))
+  .views((self) => ({
+    get isLoading() {
```
Comment (PAKovalev) [inline src/core/models/OnlineUsersList/OnlineUsersList.ts:40]: `get isLoading(): boolean {`

File: src/scenes/widgets/pages/ProfileWidget/components/templates/MyProfileEditor/MyProfileEditor.styled.ts [hunk 6]
```diff
@@ -20,6 +20,13 @@ export const AvatarContainer = styled.div`
   flex-shrink: 0;
 `;
 
+export const ImagePreview = styled.img`
+  width: 78.22px;
```
Comment (PAKovalev) [inline src/scenes/widgets/pages/ProfileWidget/components/templates/MyProfileEditor/MyProfileEditor.styled.ts:24]: 78

File: src/scenes/widgets/pages/ProfileWidget/components/templates/MyProfileEditor/MyProfileEditor.tsx [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+    if (!userProfile?.profile) {
+      return;
+    }
+
+    setValue('name', userProfile.name);
+    setValue('profile', userProfile.profile);
+  }, [userProfile?.name, userProfile?.profile, setValue]);
+
+  useEffect(() => {
+    UnityService.setKeyboardControl(false);
```
Comment (PAKovalev) [inline src/scenes/widgets/pages/ProfileWidget/components/templates/MyProfileEditor/MyProfileEditor.tsx:45]: unityService

File: src/scenes/widgets/stores/ProfileStore/ProfileStore.ts [hunk 8]
```diff
@@ -83,13 +93,47 @@ const ProfileStore = types.compose(
...
+      }),
+      setImage(image?: File) {
+        self.selectedImage = image;
+      },
+      openEdit() {
+        self.isEditingProfile = true;
+      }
+    }))
+    .views((self) => ({
+      get fetchSpaceIsDone() {
```
Comment (PAKovalev) [inline src/scenes/widgets/stores/ProfileStore/ProfileStore.ts:128]: `: boolean`

**Review by PAKovalev — CHANGES_REQUESTED** [review]

**Review by PAKovalev — APPROVED** [review]

**Review by Daboo1998 — APPROVED** [review]

**Pull request merged**